### ISSUE TITLE
New to this, but want to contribute in the data-structures area. 

### ISSUE THREAD

**Issue by GregRos** [fsprojects/FSharpx.Extras#207]
A while ago I became interested in persistent data structures and implemented several of them in F# (original implementation, not a port, though almost none of them are original themselves). In some cases, I also wrote some text explaining exactly how they worked and why they do. I've since abandoned the project, but I've looked at it again and I don't want that work to go to waste. 

The explanations have colorful pictures too. I was inspired by the infamous Haskell book.

Some of the data structures are a real-time queue/deque that doesn't use laziness, a HAMT-based hash map with a few tricks like path compression, a HAMT-based vector, a finger tree, an indexed list...

They're all written with clarity and readability in mind.

They're not ports, so they're written for F#.

I wouldn't use them as they are (some aren't fully finished, and enough time has passed that I'm not certain which is which). However, I would like to contribute in this area somehow. There are significant portions that are reusable.

I'm rather new to this (meaning, open source stuff, github, etc), so I don't know exactly how things work, what to do, or what needs to be done.

I've also had some ideas about design-related things and I'm not sure what to do with them. 

In short, I'm in desperate need of some direction. Thank you :)
